### PR TITLE
executor: migrate test infra to testify for concurrent_map_test.go and collation_test.go

### LINKED ISSUE #28570: migrate test-infra to testify for `collation_test.go`

### PR DESCRIPTION
**Pull request by zzl200012**
### What problem does this PR solve?

Issue Number: close #28570 #28571 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/concurrent_map_test.go [hunk 1]
```diff
@@ -16,19 +16,20 @@ package executor
...
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/util/chunk"
+	"github.com/stretchr/testify/require"
 )
 
 // TestConcurrentMap first inserts 1000 entries, then checks them
-func (cm *pkgTestSuite) TestConcurrentMap(c *C) {
+func TestConcurrentMap(t *testing.T) {
 	m := newConcurrentMap()
```
Comment (tisonkun) [inline executor/concurrent_map_test.go:27]:
```suggestion
func TestConcurrentMap(t *testing.T) {
	t.Parallel()
	m := newConcurrentMap()
```

Run in parallel in a `xxx_test.go`. Also this test seems can be run in parallel.

**Review by tisonkun — APPROVED** [review]
LGTM.

/cc @xhebox @tiancaiamao

File: executor/concurrent_map_test.go (hunk 1 above)
Comment (xhebox) [reply executor/concurrent_map_test.go:27]: Wait for this change :)

Comment (zzl200012) [reply executor/concurrent_map_test.go]: done

Comment (tisonkun) [reply executor/concurrent_map_test.go]: @xhebox @tiancaiamao could you please give it another round to review?

**Review by disksing — APPROVED** [review]

**Pull request merged**